torch_np/_detail/implementations.py

```diff
@@ -450,6 +462,10 @@ def indices(dimensions, dtype=int, sparse=False):
 
 
 def bincount(x_tensor, /, weights_tensor=None, minlength=0):
+    if x_tensor.numel() == 0:
+        # edge case allowed by numpy
+        x_tensor = torch.as_tensor([], dtype=int)
```

Comment:
`x = x.new_empty(()) `may do? This way if `x_tensor` was of an incorrect dtype we'll error out as well.

Reply:
Don't think so: 

```
(Pdb) x.new_empty((), dtype=int)
tensor(140087024053344)
```

has `.numel() == 1` instead of zero.

Reply:
Sorry, I meant `x = x.new_empty(0)`

Reply:
Ah, right. Did not know about `new_empty` before. Thanks. 
Still needs a `new_empty(0, dtype=int)` for safe casting from the default float64 to integer; reads way better anyway. Thank you.

Reply:
If you don't need the dtype/device/layout of `x`, then you are probably better off calling `torch.empty`.

This reminds me that, once we try things with different dtypes, we'll probably need to be careful when using constructors, as calling `torch.empty` will create a tensor in CPU, and we probably want to maintain the device of the input tensor in our intermediary tensors. Now, this is just me thinking out loud. Nothing t do here.